### DISCUSSION TITLE
Board only works when USB cable is connected

### DISCUSSION THREAD

**Discussion by nickmachielse** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#514] [Q&A]
Hi,

I am facing a weird issue. I have programmed the board and connected it to my Helsinki pre2021 pump. I am able to connect to WiFI and setup the hardware. The problem is that the pump display is not working and buttons do not work as well. Also I cannot control the pump through the web interface. Now when I connect a USB cable to the ESP and start monitoring from PlatformIO somehow everything works fine. Even when disconnecting USB cable and resetting the ESP display and buttons are working correctly. 
When disconnecting power from mains and enable power again I end up with the same problem.
Has anyone seen this before and can help me fixing this?
I am using the blue level converter and 10K resistor.

**Comment by visualapproach** [comment]
I've seen similar report but it's weird. Maybe not enough power during start, or grounding through the USB may do something.
What happens if you reset the ESP when it doesn't work?
What happens if you start the pump with the usb metal shield touching the ESP USB shield (Connecting the GND to the computer)
Does it help cutting the wires shorter?

**Comment by nickmachielse** [comment]
Thanks!

Resetting the ESP when it does not work does not help. Also enabling pump
power with USB connected to PC does not help.
The only situation to get it working is to start Monitoring task from
PlatformIO.
I can try to cut the wires shorter but still puzzles me that somehow
monitoring session resolves the issue. Is there a way to get debug log file
without USB being connected?

**Reply by visualapproach** [reply]
Ok what if you have usb connected and you reset the esp by pressing the little reset button? I can't understand what the serial monitor would do to change things, except that it resets the esp when starting.

**Comment by nickmachielse** [comment]
In that case everything works fine. Even when disconnecting USB it keeps
working, also after resetting the board. Only when reconnecting pump power
it is down again.

**Comment by visualapproach** [comment]
You can check bootlog.txt

**Comment by nickmachielse** [comment]
In bootlog I see only External System and Power On messages, no errors.

**Comment by visualapproach** [comment]
It smells like a power issue during pump start up. Strange though that a reset won't fix it without usb connected. If the esp takes too much power during startup it shouldn't work after rebooting without the usb. 
It doesn't make sense but I have a few more 'what ifs' 
What if you plug in the esp whilst pump is on? Well that's a bad test. We usually don't want to disconnect when powered.
What if you power the esp with a powerbank and reset it after pump start?

**Comment by nickmachielse** [comment]
I tried all scenarios but none of them worked.
I did notice that when pressing these 4 buttons I get to hear some sound (
nothing happens though ).
- Power
- Temp +
- Temp -
- Clock

**Reply by visualapproach** [reply]
You can log messages with these addresses
Layzspa.local/debug-on/
Layzspa.local/debug-off/

Data will be saved in log.txt

**Comment by nickmachielse** [comment]
It seems that communication fails, see screenshot...

I have tried older versions too without lock.
The only other option I could think of is too build another PCB, I do have
a spare one.

Please let me know if something else comes to your mind 🙏.


Op vr 19 mei 2023 14:30 schreef visualapproach ***@***.***>:

**Reply by visualapproach** [reply]
I don't see a screenshot 😎

**Comment by nickmachielse** [comment]
Oops...
![Screenshot_20230519_192019_Firefox](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/134002626/8a1f535c-95e1-43b0-bb7f-5d9e34e72e02)

**Reply by visualapproach** [reply]
The display sends button presses alright it seems. But nothing from the cio. I'd cut the wires short and try again.

**Reply by HankM34** [reply]
I have got similar problem, NodeMCUv2, TXS0108. Lay-Z-Spa Hawaii about 5 years old. With version 2.1.6 (model NO54154) I have communication with pump and display, I am able to use it. But with newer versions it is not working at all. I have tested versions between 3.5.0 up to 4.1.3. I have checked log.txt file and I have same problem as @nickmachielse, I am able to communicate with display but not with pump. Any suggestions? 

Thanks

Hank

**Reply by visualapproach** [reply]
@HankM34 so v2.1.6 is latest working code?

**Comment by chunkysteveo** [comment]
My 4wire that doesn't work on the newer firmware is also the NO54154... coincidence or root issue?

**Reply by visualapproach** [reply]
With the difference that your worked with v3.5.0 right?!

**Reply by HankM34** [reply]
@visualapproach I have made few tests and here you have results:

3.0.0 OK
3.1.1 OK
3.2.0 OK
3.3.0 OK
3.4.0 OK
3.5.0 failed compilation
3.5.1 pump is passing info to control panel, panel is working but is not possible to take over control and do change to any setting
4.0 alpha is crashing after 54154 setup in web menu.
4.0 beta pump is passing info to control panel, panel is working but is not possible to take over control and do change  to any setting
4.1.3 there is no communication at all, not possible to control with panel or over the web gui.

**Reply by visualapproach** [reply]
Just to make it clear to me:
@chunkysteveo has a 54154, 4-wire.
@nickmachielse has a 2021 6-wire.
@HankM34 has a 54154 4-wire I assume, judging from your last post.

@nickmachielse 's problem is mainly that it doesn't start working without USB connected + reboot.
The 54154 has communication problems with the pump, that is NOT solved with USB+reboot?

**Reply by HankM34** [reply]
@visualapproach Hi, I have forgot to mention that is 4 wire version.

**Reply by nickmachielse** [reply]
I my case it only seems to work when starting monitoring session from PlatformIO. After that it keeps running fine, even with ESP reset. 
I tried using shorter wires, without luck. I also tried older versions starting from v3,5,1 till latest. All same issue.
I will create a new PCB with new components and try again later...

**Reply by chunkysteveo** [reply]
I think @HankM34  and I have similar/same issue. But different to @nickmachielse and his USB type issue? Perhaps we peel off this issue @HankM34 and move to the one I have open about the 4 wire NO54154 pump? What you think guys?

**Comment by bertieuk** [comment]
So I’m experiencing this issue as well. I have a home built board and been running v1.something for last few years. 

Got my tub out a few weeks ago for the summer but wouldn’t come on. So thought I’d upgrade the firmware. Realised it’s moved from Arduino to Visual Studio, tried latest version and all the way down to one that would compile and flash, and 2.16 seems to have worked. But will only work with usb power provided. (Usb power bank in photo)

![IMG_7559](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/13354213/7690904a-3572-40b7-82b5-9cb1132a911e)

**Reply by visualapproach** [reply]
If it didn't come on before the update, don't you think there was something wrong with the device, rather than the firmware?
